### PR TITLE
Renamed Template to TemplateRenderer

### PR DESCRIPTION
**Pull request by ojhaujjwal**
As discussed in #136, I have renamed all the occurance of `Template` to `TemplateRenderer` and `$template` to `$renderer`

### CODE REVIEW

File: src/Template/Plates.php [hunk 1]
```diff
@@ -15,7 +15,7 @@
 /**
  * Template implementation bridging league/plates
  */
-class Plates implements TemplateInterface
+class Plates implements TemplateRendererInterface
```
Comment (bakura10) [inline src/Template/Plates.php]: Similarily to Route, I think we should suffix by Renderer (PlatesRendere, ZendRenderer, TwigRenderer...)

File: doc/book/template/interface.md [hunk 2]
```diff
@@ -71,7 +71,7 @@ more paths to templates; these are then used when resolving a template name to
 the actual template. You may use the `addPath()` method to do so:
 
 ```php
-$template->addPath('templates');
+$templateRenderer->addPath('templates');
 ```
```
Comment (weierophinney) [inline doc/book/template/interface.md]: Too verbose; please use `$renderer` instead.

File: src/Template/Plates.php (hunk 1 above)
Comment (ojhaujjwal) [inline src/Template/Plates.php]: Agree

File: doc/book/template/interface.md [hunk 3]
```diff
@@ -1,13 +1,13 @@
 # The Template Interface
```
Comment (weierophinney) [inline doc/book/template/interface.md]: Note to self: update this title to "The Template Renderer Interface" on merge.

**Pull request merged**